### ISSUE TITLE
Node-red-node-twitter retweet support

### ISSUE THREAD

**Issue by baruchiro** [node-red/node-red-nodes#834]
Hi, the `node-red-node-twitter` is not supporting retweets, since it uses [`/statuses/update.json`](https://github.com/node-red/node-red-nodes/blob/96fa500ad663f68d49f456036421994d4ab25cf8/social/twitter/27-twitter.js#L640), and the retweet is done by [`statuses/retweet/:id`](https://developer.twitter.com/en/docs/twitter-api/v1/tweets/post-and-engage/api-reference/post-statuses-retweet-id).

I want to create a PR to add this support. I think since it is a different interface- it needs only the tweet id to retweet it, I will create a new node for it.

Or maybe the preferred way is to implement it inside the Twitter Out node?

WDYT?

I asked [in the forum](https://discourse.nodered.org/t/node-red-node-twitter-retweet-support/50038?u=baruchiro), but no answers there.

**Comment by dceejay** [comment]
I think it would be cleaner as part of the existing node - but have not thought in detail about the implications of that.

**Comment by knolleary** [comment]
Agreed it would be better to have this in the existing output node.

There are lots of features of the Twitter API we don't support today, so it would be good to keep that in mind when adding individual features.

For this, I'd suggest if the received message includes `msg.retweet` then use its value as the tweet id to retweet.

The app lets you retweet with comment - but that doesn't appear to be part of the `/status/retweet/` endpoint. It would make sense if `msg.payload` is set to some text as well as `msg.retweet` set for that to be a retweet with comment - its just a matter of figuring out the right API to use for that.

@baruchiro you are most welcome to explore this and propose something along these lines.

**Comment by baruchiro** [comment]
OK, about retweet with text, I'm not sure I will take this. Maybe if it will be on my way.

About the existing output node. First, I agree. But the reason I was thinking to implement a new output node is in order to support the simplest flow, input to output:

![image](https://user-images.githubusercontent.com/17686879/132747095-f9dd029e-6733-4691-8b5c-b3ab8d942178.png)

Today, if you are wiring such flow, it will tweet the input tweet, as **text** (or any other content) but not like a real **retweet** supported by Twitter.

With implementing in the existing node, this simple flow will require another node to take the tweetID and put it in the payload.

**Comment by moshed** [comment]
was this ever updated to support retweets? I tried using `msg.retweet` with no success
